java-source/src/main/java/net/corda/training/state/IOUState.java

```diff
@@ -48,38 +51,48 @@ public IOUState(Amount<Currency> amount, Party lender, Party borrower) {
     }
 
     public Party getLender() {
-        return lender;
+	    return lender;
     }
 
     public Party getBorrower() {
         return borrower;
     }
 
-    public Amount<Currency> getPaid() {
-        return paid;
+    public Amount getPaid() {
+	    return paid;
     }
 
     @Override
-    public List<AbstractParty> getParticipants() {
-        return ImmutableList.of(lender, borrower);
+    public UniqueIdentifier getLinearId() {
+	    return linearId;
     }
 
-    @Override
-    public UniqueIdentifier getLinearId() {
-        return linearId;
+    /**
+     *  This method will return a list of the nodes which can "use" this state in a valid transaction. In this case, the
+     *  lender or the borrower.
+     */
+   	@Override
+    public List<AbstractParty> getParticipants() {
+        return ImmutableList.of(borrower, lender);
     }
 
     /**
      * Helper methods for when building transactions for settling and transferring IOUs.
      * - [pay] adds an amount to the paid property. It does no validation.
      * - [withNewLender] creates a copy of the current state with a newly specified lender. For use when transferring.
+     * - [copy] creates a copy of the state using the internal copy constructor ensuring the LinearId is preserved.
      */
-    public IOUState pay(Amount<Currency> amountToPay) {
-        Amount<Currency> newAmountPaid = this.paid.plus(amountToPay);
+    public IOUState pay(Amount<Currency> paidAmount) {
+        Amount<Currency> newAmountPaid = this.paid.plus(paidAmount);
         return new IOUState(amount, lender, borrower, newAmountPaid, linearId);
     }
 
     public IOUState withNewLender(Party newLender) {
         return new IOUState(amount, newLender, borrower, paid, linearId);
     }
+
+    public IOUState copy(Amount<Currency> amount, Party lender, Party borrower, Amount<Currency> paid) {
+        return new IOUState(amount, lender, borrower, paid, this.getLinearId());
```

Comment:
Java has a copy method you may want to leverage as a best practice.

Reply:
Java has a clone() method but requires the class to implement the @clonable annotation and is (according to stack overflow) not best practice. I think the private copy constructor method is generally considered the best way to implement a custom copy method, which in our case preserves the LinearId while allowing other properties to be changed.